File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const mymap=new Map();

    for(let i=0; i<nums.length; i++){
        let comp=target-nums[i];
        
        if (mymap.has(comp)){
            let a=[mymap.get(comp), i];
            return a;
        }

        mymap.set(nums[i], i);

    }
};

File: 1-two-sum/two-sum.cpp

class Solution {
public:
    vector<int> twoSum(vector<int>& arr, int target) {
      unordered_map<int, int>m;
      vector<int>v(2, 0);
      for(int i=0; i<arr.size(); i++){
        int comp=target-arr[i];
        if(m.count(comp)){
            return {m[comp], i};
        }
        m[arr[i]]=i;
      }
      
      return v;
    }
};